spec/models/course/condition/assessment_spec.rb

```diff
@@ -0,0 +1,99 @@
+      context 'when there is no minimum grade percentage' do
+        it 'returns the correct title' do
+          subject.assessment = create(:course_assessment_assessment, title: 'Dummy', course: course)
+          expect(subject.title).to eq('Dummy')
+        end
+      end
+
+      context 'when there is minimum grade percentage' do
+        it 'returns the correct title' do
+          pending('Assessment#title uses number_to_percentage which uses I18 but we change I18
+behaviors in our test environment and this test will throw an exception. It works
+fine when the change is disabled.')
+          subject.assessment = create(:course_assessment_assessment, title: 'Dummy', course: course)
+          subject.minimum_grade_percentage = 33.333
+
+          expect(subject.title).to eq('Dummy (more than 33.33%)')
+        end
+      end
+    end
+
+    describe '#satisfied_by?' do
+      let(:course_user) { create(:course_user) }
+      let(:assessment) do
+        # 10 questions, 2 marks each => maximum grade is 20
+        assessment = create(:assessment)
+        create_list(:course_assessment_question_multiple_response, 10, assessment: assessment)
+        assessment
+      end
+
+      context 'when there is not minimum grade percentage' do
+        subject do
+          condition = create(:course_condition_assessment)
+          condition.assessment = assessment
+          condition
+        end
+
+        it 'return false if there is no submission' do
+          expect(subject.satisfied_by?(course_user)).to be_falsey
+        end
+
+        it 'return false if the submission is not graded' do
+          create(:submission, workflow_state: :attempting, assessment: assessment,
+                              user: course_user.user)
+          expect(subject.satisfied_by?(course_user)).to be_falsey
+        end
+
+        it 'return true if the submission is graded' do
+          create(:submission, workflow_state: :graded, assessment: assessment,
+                              user: course_user.user)
+          expect(subject.satisfied_by?(course_user)).to be_truthy
+        end
+      end
+
+      context 'when there is minimum grade percentage' do
+        subject do
+          condition = create(:course_condition_assessment, minimum_grade_percentage: 60)
+          condition.assessment = assessment
+          condition
+        end
+        let(:submission) do
+          create(:submission, workflow_state: :graded, assessment: assessment,
+                 user: course_user.user)
+        end
+
+        it 'return false if grade is below the minimum grade percentage' do
+          answers = assessment.questions.attempt(submission)
+          answers.each do |answer|
+            answer.finalise!
+            answer.grade = 1
+            answer.save!
+          end
+
+          expect(subject.satisfied_by?(course_user)).to be_falsey
+        end
+
+        it 'return true if grade is at least the minimum grade percentage' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.